docs/source/reference/ha.rst

```diff
@@ -0,0 +1,345 @@
+If content folders i.e. ``/opt/stackstorm/packs`` and ``/opt/stackstorm/virtualenvs`` are placed on read-write NFS
+mounts then writing from any |st2| node will be visible to other nodes. Special care needs to be take in case
+of ``/opt/stackstorm/virtualenvs`` since that has symlinks to system libraries. If care is not taken to provision
+all host boxes in an identical manner it could leads to unpredicatble behavior. Although possible to implement in
+this manner it is certainly not ideal and perhaps managing the ``virtualenvs`` on every host box individually would
+be a more robust approach.
+
+Content management
+^^^^^^^^^^^^^^^^^^
+Managing pack installation using a content management tool of your choice. Assuming that the list of packs to be deployed
+will be static in deployments then deploying content to |st2| nodes via CM tools could be a sub-step of an overall
+|st2| deployment. This is perhaps the better of the two approaches to end up with a predicatble HA deployment of |st2|.
+
+Reference HA setup
+------------------
+
+In this section we provide a highly opinionated and therefore prescriptive approach to deploying |st2| in HA. This deployment
+has 3 independent boxes which we categorize as controller box and blueprint box. Let call these boxes ``st2-multi-node-cntl``,
+``st2-multi-node-1`` and ``st2-multi-node-2``. For the sake of reference we will be using Ubuntu 14.04 as the base operating
+system.
+
+.. figure :: /_static/images/st2-deployment-multi-node.png
+    :align: center
+
+    StackStorm HA reference deployment.
+
+.. source https://docs.google.com/drawings/d/1_BJa9ZtBjFa1Dxx6cPiFlmpTS9AsNzkkvp_vuyVV3bw/edit
+
+Controller box
+^^^^^^^^^^^^^^
+This box runs all the shared required dependencies and some |st2| components.
+
+* Nginx as loadbalancer
+* MongoDB
+* PostgreSQL
+* RabbitMQ
+* st2chatops
+* st2web
+
+In practice ``MongoDB``, ``PostgreSQL`` and ``RabbitMQ`` are often in standalone clusters managed outside of |st2|.
+The 2 shared components i.e. ``st2chatops`` and ``st2web`` are placed here for sake of convenience and could be placed anywhere
+with the right configuration.
+
+Nginx acting as the loadbalancer can easily be switched out for Amazon ELB, HAProxy or any other of your choosing. In that case
+``st2web`` which is being served off this Nginx will also need a new home.
+
+``st2chatops`` which use ``hubot`` is not easily deployed in HA. Using something like `keepalived <http://www.keepalived.org/>`__
+to maintain st2chatops in active-passive configuration would be an option.
+
+Following are the steps to provision a controller box on Ubuntu 14.04.
+
+Install required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Install ``MongoDB``, ``PostgreSQL`` and ``RabbitMQ``.
+
+  .. code-block:: bash
+
+        sudo apt-get install -y mongodb-server rabbitmq-server postgresql
+
+
+2. Fix listen address in `/etc/postgresql/9.3/main/postgresql.conf` and have PostgreSQL listen an interface that has an
```

Comment:
also need to do this for mongodb in `/etc/mongodb.conf` (binds to `127.0.0.1` by default, at least with 2.4.9 on ubuntu 14.04)